tools/validation-generator/src/main/java/io/spine/tools/validate/ExternalConstraintFlag.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.validate;
+
+import com.google.common.base.Objects;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.FieldSpec;
+import io.spine.code.proto.FieldDeclaration;
+import io.spine.tools.validate.code.BooleanExpression;
+import io.spine.tools.validate.code.Expression;
+import io.spine.validate.ExternalConstraints;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+import static javax.lang.model.element.Modifier.FINAL;
+import static javax.lang.model.element.Modifier.PRIVATE;
+import static javax.lang.model.element.Modifier.STATIC;
+
+/**
+ * A boolean flag in the generated code which signifies whether or not a field has externally
+ * defined constraints.
+ *
+ * <p>Such a flag is only generated for message fields marked with {@code (validate) = true}. Other
+ * fields cannot be target for external constraints.
+ */
+final class ExternalConstraintFlag {
+
+    private final FieldDeclaration declaration;
+    private final BooleanExpression field;
+
+    ExternalConstraintFlag(FieldDeclaration field) {
+        this.declaration = checkNotNull(field);
+        String name = format("is%sValidatedExternally", field.name().toCamelCase());
+        this.field = BooleanExpression.fromCode(name);
+    }
+
+    /**
+     * Generated an expression which obtains the value of this flag.
+     *
+     * <p>The value of the flag must be set when this expression is used. Otherwise,
+     * a {@code NullPointerException} is thrown.
+     *
+     * @return an expression of the primitive value of the flag
+     */
+    BooleanExpression value() {
+        return field;
+    }
+
+    /**
+     * Obtains this flag as a {@link ClassMember}.
+     */
+    ClassMember asClassMember() {
+        ClassName containingTypeName = ClassName.bestGuess(declaration.declaringType()
```

Comment:
As the lookup seems expensive, I would cache the result. Or compute it right away in the ctor.